test/reporter_test.rb

```diff
@@ -1,30 +1,55 @@
 require 'minitest/autorun'
+require 'builder'
 require 'stringio'
 require 'time'
 
 require 'minitest/junit'
 
 class ReporterTest < Minitest::Test
+  class MockResult
```

Comment:
Use `Minitest::Result` instead

Reply:
Agreed, reworked as requested to use the Minitest::Result object.